devito/passes/iet/linearization.py

```diff
@@ -129,14 +125,25 @@ def __init__(self, mode, dtype, sregistry):
         self.strides_dynamic = {}  # Strides varying regularly across iterations
         self.dists = {}
 
+    def _select_indexing_dtype(self, f):
+        # Some objects may contain an extremely large number of elements, so we
+        # conservatively use int64 to avoid potential overflows regardless of
+        # what the user requested via `index-mode`
+        if f.is_SparseTimeFunction:
```

Comment:
Just `SparseFunction` I would say, given that the issue occurs due to a large number of points, rather than a large nt iirc

Reply:
order of millions is fine, order of billions (due to time...) isn't

Reply:
Sure, but those are different array dimensions right? Unless the array is linearised?